### PR TITLE
Coroutine executors

### PR DESCRIPTION
**Pull request by HeyAlex**
coroutine executors (pure kotlin and android)

### CODE REVIEW

File: executor/executor-coroutine-android/build.gradle [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    sourceSets {
+        main.java.srcDirs += "src/main/kotlin"
+        test.java.srcDirs += "src/test/kotlin"
+    }
+
+}
+
+dependencies {
+    implementation project(':core')
+    api project(':executor:executor-coroutine')
```
Comment (Unlimity) [inline executor/executor-coroutine-android/build.gradle]: Are you sure that `api` is fine? In this case if user will import only `core` and `executor-coroutine-android` is probably won't work

File: executor/executor-coroutine/src/main/kotlin/com/agoda/boots/executor/CoroutineExecutor.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package com.agoda.boots.executor
+
+import com.agoda.boots.Executor
+import com.agoda.boots.Executor.Companion.DEFAULT_CAPACITY
+import kotlinx.coroutines.*
+
+/**
+ * Coroutine implementation of [Executor].
+ *
+ * Can switch to dispatcher to invoke listeners
```
Comment (Unlimity) [inline executor/executor-coroutine/src/main/kotlin/com/agoda/boots/executor/CoroutineExecutor.kt]: There is something wrong with this description. I think it uses dispatcher to launch concurrent bootables only, for listeners and non-concurrent bootables dispatcher is not used in your implementation.

**Comment by Unlimity** [comment]
#4

File: executor/executor-coroutine-android/build.gradle (hunk 1 above)
Comment (Unlimity) [reply executor/executor-coroutine-android/build.gradle]: My bad here. Revised gradle docs and you're more that right to use `api` here. 👍

File: executor/executor-coroutine/src/main/kotlin/com/agoda/boots/executor/CoroutineExecutor.kt (hunk 2 above)
Comment (HeyAlex) [reply executor/executor-coroutine/src/main/kotlin/com/agoda/boots/executor/CoroutineExecutor.kt]: Fixed

**Comment by Unlimity** [comment]
Great! Thanks for the submission!

**Pull request merged**